lumicks/pylake/nb_widgets/kymotracker_widgets.py

```diff
@@ -140,49 +140,25 @@ def set_xlim(drag_event):
 
         MouseDragCallback(self._axes, 1, set_xlim)
 
-    def _get_scale(self):
-        """Get scaling of the image axes"""
-        return tuple(lims[1] - lims[0] for lims in (self._axes.get_xlim(), self._axes.get_ylim()))
-
     def _connect_line_callback(self):
         canvas = self._axes.figure.canvas
+        scale = tuple(lims[1] - lims[0] for lims in (self._axes.get_xlim(), self._axes.get_ylim()))
         cutoff_radius = 0.05  # We use a connection cutoff of 5% of the axis ranges
         clicked_line_info = None
         plotted_line = None
         nodes = None
 
-        def get_node_info():
-            nodes = [
-                np.vstack(
-                    (
-                        np.full(len(track), j),  # track index
-                        np.arange(len(track)),  # node index within track
-                        track.seconds,  # x-coordinate
-                        track.position,  # y-coordinate
-                    )
-                )
-                for j, track in enumerate(self.lines)
-            ]
-            return np.hstack(nodes).T
-
-        def get_nearest(x, y):
-            nonlocal nodes
-            ref_position = np.array([x, y])
-            squared_dist = np.sum(((ref_position - nodes[:, -2:]) / self._get_scale()) ** 2, 1)
-            idx = np.argmin(squared_dist)
-            return np.sqrt(squared_dist[idx]), idx
-
         def initiate_line(event):
             nonlocal nodes, clicked_line_info
             if len(self.lines) == 0:
                 return
 
-            nodes = get_node_info()
-            distance, idx = get_nearest(event.x, event.y)
-
-            if distance < cutoff_radius:
-                line_index, *line_info = nodes[idx]
-                clicked_line_info = [self.lines[int(line_index)], *line_info]
+            nodes = _get_node_info(self.lines)
```

Comment:
It's not a problem that I now have to call `_get_node_info()` myself, but it does leak implementation details that we don't really need, thus binding you a bit to this admittedly still private API.

All _I_ need, for the `_merge_lines()` to work, is the position of the datapoint inside a `KymoLine`. Maybe we can rethink this part a bit.

Reply:
I agree, the API was a bit weird. basically I left it as it already was, but just pulled out into a separate function. I've updated it now to be a bit more straightforward I think. `_get_node_info()` is back to being truly internal, and `_get_nearest()` returns a `dict` with the relevant info if a node was clicked, else empty.

See last 3 commits for proposed changes, these'll be squashed on merge